### PR TITLE
large language model interface

### LINKED ISSUE #1034: create language model interface that caches responses to disk
goal: never waste money querying the same LLM with the same prompt twice

cc @NishanthJKumar

### PR DESCRIPTION
**Pull request by tomsilver**
closes #1034

### CODE REVIEW

File: src/llm_interface.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+import openai
+
+from predicators.src.settings import CFG
+
+# This is a special string that we assume will never appear in a prompt, and
+# which we use to separate prompt and completion in the cache. The reason to
+# do it this way, rather than saving the prompt and responses separately,
+# is that we want it to be easy to browse the cache as text files.
+_CACHE_SEP = "\n####$$$###$$$####$$$$###$$$####$$$###$$$###\n"
```
Comment (NishanthJKumar) [inline src/llm_interface.py:16]: smart 🧠

File: src/llm_interface.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                           prompt: str,
+                           temperature: float,
+                           seed: int,
+                           num_completions: int = 1) -> List[str]:
+        """Sample one or more completions from a prompt.
+
+        Higher temperatures will increase the variance in the responses.
+
+        The seed may not be used, and the results may therefore not be
+        reproducible, for LLMs where we only have access through an API that
```
Comment (NishanthJKumar) [inline src/llm_interface.py:51]: nit: get rid of the comma on this line

**Review by NishanthJKumar — APPROVED** [review]
LGTM! Thanks for the quick and thorough work on this 🚀

**Comment by ronuchit** [comment]
i love this interface! one comment: could you update the docstring of `_sample_completions` to clarify that this is the main method that subclasses implement by invoking an LLM

**Comment by tomsilver** [comment]
https://github.com/Learning-and-Intelligent-Systems/predicators/pull/1072

**Pull request merged**